### ISSUE TITLE
Rule Request: ARG parameters used and defined out of order

### ISSUE THREAD

**Issue by hype8912** [hadolint/hadolint#1075]
When using docker if you define an ARG parameter before the FROM [image] line it's only available to the FROM line. Attempting to use it anywhere after the FROM will show as an empty value.

The rule request is if an $ARGUMENT is used after the FROM but defined before the FROM that it gives an error. In the Bad example below TWINE_MIN_VERSION would never be available. I know there are other ways to do this, it was just a simple example file I had to use for a demonstration.

Vice versa for if an ARG parameter is used in the FROM line but defined after the FROM line that it should give an error.

Example Good (before):

```
ARG IMAGE_VERSION

FROM python:${IMAGE_VERSION}

RUN apk update && \
    apk add ca-certificates gnupg && \
    rm -rf /var/cache/apk/*

RUN pip install twine>=6.0.0

ENTRYPOINT ["twine"]
```

Example Bad (before):

```
FROM python:${IMAGE_VERSION}

ARG IMAGE_VERSION

RUN apk update && \
    apk add ca-certificates gnupg && \
    rm -rf /var/cache/apk/*

RUN pip install twine>=6.0.0

ENTRYPOINT ["twine"]
```

Example Good (after):

```
FROM python:3-alpine

ARG TWINE_MIN_VERSION

RUN apk update && \
    apk add ca-certificates gnupg && \
    rm -rf /var/cache/apk/*

RUN pip install twine>="${TWINE_MIN_VERS}"

ENTRYPOINT ["twine"]
```

Example Bad (after):

```
ARG TWINE_MIN_VERSION

FROM python:3-alpine

RUN apk update && \
    apk add ca-certificates gnupg && \
    rm -rf /var/cache/apk/*

RUN pip install twine>="${TWINE_MIN_VERS}"

ENTRYPOINT ["twine"]
```
